### DISCUSSION TITLE
Problem with foundry tests

### DISCUSSION THREAD

**Discussion by MostPow3rful** [Cyfrin/foundry-full-course-cu#2599] [Q&A]
GM
---
Code :
```js
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.18;

import {Test} from "forge-std/Test.sol";
import {console} from "forge-std/console.sol";
import "@openzeppelin/contracts/token/ERC20/IERC20.sol";

contract TestERC20_2 is Test {
    address AAVE_CONTRACT_ADDRESS;

    address AAVE_USER_HOLDER;

    IERC20 aaveToken;

    function setUp() public {
        AAVE_CONTRACT_ADDRESS = vm.parseAddress(
            "0x7fc66500c84a76ad7e9c93437bfc5ac33e2ddae9"
        );

        AAVE_USER_HOLDER = vm.parseAddress(
            "0x2efb50e952580f4ff32d8d2122853432bbf2e204"
        );
        
        aaveToken = IERC20(AAVE_CONTRACT_ADDRESS);

        console.log(
            "Balance Of [AAVE HOLDER] [TOKEN]: %s",
            aaveToken.balanceOf(AAVE_USER_HOLDER)
        );
    }
}
```
***
Error :
```js
[FAIL. Reason: setup failed: EvmError: Revert] setUp() (gas: 0)
Traces:
  [56172] TestERC20_2::setUp()
    ├─ [0] VM::parseAddress("0x7fc66500c84a76ad7e9c93437bfc5ac33e2ddae9") [staticcall]
    │   └─ ← [Return] InitializableAdminUpgradeabilityProxy: [0x7Fc66500c84A76Ad7e9c93437bFc5Ac33E2DDaE9]
    ├─ [0] VM::parseAddress("0x2efb50e952580f4ff32d8d2122853432bbf2e204") [staticcall]
    │   └─ ← [Return] 0x2eFB50e952580f4ff32D8d2122853432bbF2E204
    ├─ [0] InitializableAdminUpgradeabilityProxy::balanceOf(0x2eFB50e952580f4ff32D8d2122853432bbF2E204) [staticcall]
    │   └─ ← [Stop]
    └─ ← [Revert] EvmError: Revert

Suite result: FAILED. 0 passed; 1 failed; 0 skipped; finished in 670.80µs (0.00ns CPU time)

Ran 1 test suite in 7.89ms (670.80µs CPU time): 0 tests passed, 1 failed, 0 skipped (1 total tests)

Failing tests:
Encountered 1 failing test in test/erc20-2/ERC20_2.t.sol:TestERC20_2
[FAIL. Reason: setup failed: EvmError: Revert] setUp() (gas: 0)

Encountered a total of 1 failing tests, 0 tests succeeded
```
***
i checked the code and when i remove `console.log` code, the problem will be solved ! and the next string thing i saw is that i wrote `aaveToken.balanceOf(AAVE_USER_HOLDER)` which `aaveToken` is a `IERC20` but in the traces i saw `InitializableAdminUpgradeabilityProxy::balanceOf(0x2eFB50e952580f4ff32D8d2122853432bbF2E204) [staticcall]` !!!! HOW ?

**Comment by MostPow3rful** [comment]
and i asked chatGPT. it says this address uses some proxy contract and etc. so i should to do something. i did it but not worked ( this code worked well before but now not )

**Reply by EngrPips** [reply]
Where do you get this `0x7fc66500c84a76ad7e9c93437bfc5ac33e2ddae9` from, and did you check if it is behind a proxy contract?

**Reply by MostPow3rful** [reply]
it's from another course and says it is the address of AAVE Contract. i didn't check but this code worked ok before. or i'm doing something wrong or the author of the course

**Reply by EngrPips** [reply]
Honestly, I am unsure what is wrong. Is there some forum to ask questions regarding the course? If there is, maybe you can use it.

**Reply by MostPow3rful** [reply]
yes. i think i should do this. thanks

**Reply by EngrPips** [reply]
Anytime, Friend. I will be glad to know what you find out.

**Reply by MostPow3rful** [reply]
i only must fork the rpc url with `--fork-url` and the value will be the ethereum main net

**Reply by EngrPips** [reply]
That makes sense because that contract address is on ethereum mainnet.
